lib/src/uri_template.dart

```diff
@@ -220,22 +217,21 @@ class UriTemplate {
   final String template;
   final List _parts;
 
-  UriTemplate(String template)
-      : template = template,
-        _parts = _compile(template);
+  UriTemplate(this.template) : _parts = _compile(template);
 
   String toString() => template;
 
   static UnmodifiableListView _compile(String template) {
     var parts = [];
-    template.splitMapJoin(_exprRegex,
-        onMatch: (match) { return parts.add(match); },
-        onNonMatch: (String nonMatch) {
-          if (_literalVerifier.hasMatch(nonMatch)) {
-            throw new ParseException(nonMatch);
-          }
-          if (nonMatch.isNotEmpty) parts.add(nonMatch);
-        });
+    template.splitMapJoin(_exprRegex, onMatch: (match) {
+      parts.add(match);
+      return null;
+    }, onNonMatch: (String nonMatch) {
+      if (_literalVerifier.hasMatch(nonMatch)) {
+        throw new ParseException(nonMatch);
+      }
+      if (nonMatch.isNotEmpty) parts.add(nonMatch);
+    });
```

Comment:
I'd slightly prefer to end this with an additional `,`, so that the formatting puts `onMatch` and `onNonMatch` on the same column

Reply:
Nice! TIL...